dvc/stage/__init__.py

```diff
@@ -466,6 +465,8 @@ def _checkout(out, **kwargs):
                 return None, []
             return "modified" if modified else "added", [out.path_info]
         except CheckoutError as exc:
+            if not out.store:
+                return "failed_not_stored", exc.target_infos
```

Comment:
I think we should check here that stage is not imported, out.store == False, and then just `logger.debug` the message that originally was introduced to `CheckoutError`. That way we get rid of new status for checkout, and are able to pass suggestion why we think it fails.

Reply:
We'd presumably want to set the logging level to warning, not debug, since otherwise users wouldn't see the hint unless they thought to use verbose mode. 

I agree that'll result in cleaner code, but it's a slightly coarser user experience. Instead of a nicely formatted list of files like

```
The following outputs are marked as `store: false` so remote storage was disabled, and are not present in the cache:
foo
<https://error.dvc.org/checkout-failed-no-store>
Checkout failed for following targets:
foo
Is your cache up to date?
<https://error.dvc.org/missing-files>
```

You'll get the old error message interspersed with (potentially lots) of warnings about `store: false`.

Reply:
Thanks @pared, and don't worry about the delay! 

I agree that the `failed_not_stored` addition to the statistics in `checkout` is a bit clunky, but it's the only way I could think of to provide a nice error message to the user for files with `store: false`. Seems like a shame to lose that: I don't think churning out logger messages is as nice a UX. Can you think of a way to inform `CheckoutError` objects created in `checkout` about which `outs` have `store: false` set on them without adding another category to `stats`?

Reply:
- [ ] PS wherever the warning ends up, I should add "if they're imports, try importing them" as a hint to the user